index.ts

```diff
@@ -55,63 +75,25 @@ export const onEvent: AvoInspectorPlugin['onEvent'] = async (event, { config, gl
-        eventProperties: event.properties ? convertPosthogPropsToAvoProps(event.properties) : [],
+        eventProperties: event.properties ? convertPosthogPropsToAvoProps(event.properties, global.excludeProperties, global.includeProperties) : [],
     }
 
-    try {
-        // start a tracking session
-        const sessionStartRes = await fetch('https://api.avo.app/inspector/posthog/v1/track', {
-            method: 'POST',
-            headers: global.defaultHeaders,
-            body: JSON.stringify([
-                {
-                    apiKey: config.avoApiKey,
-                    env: config.environment,
-                    appName: config.appName,
-                    createdAt: now,
-                    sessionId: sessionId,
-                    appVersion: '1.0.0',
-                    libVersion: '1.0.1',
-                    libPlatform: 'node',
-                    messageId: randomUUID(),
-                    trackingId: '',
-                    samplingRate: 1,
-                    type: 'sessionStarted',
-                },
-            ]),
-        })
-
-        if (sessionStartRes.status !== 200) {
-            throw new Error(`sessionStarted request failed with status code ${sessionStartRes.status}`)
-        }
-
-        // track events
-        const trackEventsRes = await fetch('https://api.avo.app/inspector/posthog/v1/track', {
-            method: 'POST',
-            headers: global.defaultHeaders,
-            body: JSON.stringify([avoEvent]),
-        })
-
-        // https://github.com/node-fetch/node-fetch/issues/1262
-        const trackEventsResJson = (await trackEventsRes.json()) as Record<string, any> | null
-
-        if (
-            trackEventsRes.status !== 200 ||
-            !trackEventsResJson ||
-            ('ok' in trackEventsResJson && !trackEventsResJson.ok)
-        ) {
-            throw new Error('track events request failed')
-        }
-    } catch (err) {
-        const errorMessage = err instanceof Error ? err.message : String(err)
-        console.error('Unable to send data to Avo with error:', errorMessage)
+    return {
+        url: 'https://api.avo.app/inspector/posthog/v1/track',
+        headers: global.defaultHeaders,
+        body: JSON.stringify([avoEvent]),
+        method: 'POST',
     }
 }
 
-const convertPosthogPropsToAvoProps = (properties: Record<string, any>): Record<string, string>[] => {
+const convertPosthogPropsToAvoProps = (properties: Record<string, any>, excludeProperties: Set<String>, includeProperties: Set<String>): Record<string, string>[] => {
     const avoProps = []
+
     for (const [propertyName, propertyValue] of Object.entries(properties)) {
-        if (!propertyName.startsWith("$")) {
+        const isIncluded = includeProperties.size > 0 ? includeProperties.has(propertyName) : true
+        const isExcluded = excludeProperties.has(propertyName)
+
+        if (!propertyName.startsWith("$") && isIncluded && !isExcluded) {
```

Comment:
same, let's not do anything for `$` people can choose to include/exclude them themselves

Reply:
Same, this is how the current implementation works.